Assets/MRTK/SDK/Features/UX/Scripts/Buttons/ButtonConfigHelper.cs

```diff
@@ -414,7 +414,165 @@ public void ForceRefresh()
+                return false;
+            }
+
+            string assetPath = AssetDatabase.GetAssetPath(iconQuadRenderer.sharedMaterial);
+            if (string.IsNullOrEmpty(assetPath))
+            {   // If the asset path is null, this material instance exists only in memory.
+                return false;
+            }
+
+            return true;
+        }
+
+        /// <summary>
+        /// Upgrades a button using a custom icon material.
+        /// </summary>
+        public void EditorUpgradeCustomIcon()
+        {
+            SerializedObject configObject = new SerializedObject(this);
+            SerializedProperty iconStyleProp = configObject.FindProperty("iconStyle");
+            SerializedProperty iconSetProp = configObject.FindProperty("iconSet");
+            SerializedProperty iconQuadTextureProp = configObject.FindProperty("iconQuadTexture");
+
+            if (iconQuadRenderer.gameObject.activeSelf && !iconQuadRenderer.enabled)
+            {   // If the quad renderer is disabled, enable it and disable the quad renderer game object instead.
+                // Disabling the quad renderer used to be the preferred way to disable icons but it's no longer consistent with our icon style.
+                iconQuadRenderer.enabled = true;
+                iconQuadRenderer.gameObject.SetActive(false);
+                EditorUtility.SetDirty(gameObject);
+            }
+
+            if (!iconQuadRenderer.gameObject.activeSelf && !iconSpriteRenderer.gameObject.activeSelf && !iconCharLabel.gameObject.activeSelf)
+            {   // If all the icon objects are disabled, set the icon style to none and do nothing else.
+                iconStyleProp.enumValueIndex = (int)ButtonIconStyle.None;
+                configObject.ApplyModifiedProperties();
+                EditorUtility.SetDirty(gameObject);
+                return;
+            }
+
+            string assetPath = AssetDatabase.GetAssetPath(iconQuadRenderer.sharedMaterial);
+            if (string.IsNullOrEmpty(assetPath))
+            {   // If the asset path is null, this material instance exists only in memory.
+                return;
+            }
+
+            Material targetQuadMaterial = iconQuadRenderer.sharedMaterial;
+            Texture targetQuadIcon = targetQuadMaterial.mainTexture;
+
+            if (targetQuadIcon == null)
+            {   // There is no icon to copy.
+                return;
+            }
+
+            ButtonIconSet targetIconSet = iconSet;
+            bool createdIconSet = false;
+            string generatedIconSetFolder = System.IO.Path.Combine(MixedRealityToolkitFiles.GetGeneratedFolder, customIconSetsFolderName);
+
+            // If this icon set doesn't have our icon in it, we need to either add it or create a new icon set
+            if (!iconSet.TryGetQuadIcon(targetQuadIcon.name, out Texture2D quadIcon) && iconSet == defaultIconSet)
```

Comment:
The check for ` iconSet == defaultIconSet` seems unnecessary here since in line 447 we have the following

`if (iconSet != defaultIconSet) 
{   // This button is using a custom icon set, so we can't assume material differences mean it needs an upgrade.`

In addition, we should have a field for specifying the customIconSetsFolder for the sake of testing/preventing all migrated icons from being funneled into a singular monolith icon set.

Reply:
I'll add some optional parameters you can use for testing.